lib/css/CssGenerator.js

```diff
@@ -113,14 +138,32 @@ class CssGenerator extends Generator {
 		const data = generateContext.getData();
 		data.set("css-exports", cssExportsData);
 
-		return InitFragment.addToSource(source, initFragments, generateContext);
+		const generationResult = InitFragment.addToSource(
+			source,
+			initFragments,
+			generateContext
+		);
+		if (
+			generateContext.type === "javascript" &&
+			cssModule.assertions &&
+			cssModule.assertions.type === "css"
+		) {
+			generateContext.runtimeRequirements.add(RuntimeGlobals.module);
+			return new RawSource(`const __WEBPACK_CSS_STYLE_SHEET__ =  new CSSStyleSheet();
+__WEBPACK_CSS_STYLE_SHEET__.replaceSync(${convertToTemplateLiteral(generationResult.source().toString())});${RuntimeGlobals.module}.exports = __WEBPACK_CSS_STYLE_SHEET__;`);
+		}
+		return generationResult;
 	}
 
 	/**
 	 * @param {NormalModule} module fresh module
 	 * @returns {Set<string>} available types (do not mutate)
 	 */
 	getTypes(module) {
+		const cssModule = /** @type {CssModule} */ (module);
+		if (cssModule.assertions && cssModule.assertions.type === "css") {
```

Comment:
```suggestion
		if (cssModule?.assertions?.type === "css") {
```

Reply:
It seems that this syntax is not available